packages/plugin-kafka/src/custom/admin-client.ts

```diff
@@ -106,14 +106,16 @@ export class KafkaAdminClient {
   }
 
   private async waitFor(client: KafkaClient, topicName: string, criteria: WaitCriteria, params?: retry.Options): Promise<TopicMetadata> {
-    const retryParams = merge(params, this.defaultWaitParams)
+    const retryParams = merge(this.defaultWaitParams, params)
```

Comment:
this.defaultWaitParams are going to get mutated by this

```suggestion
    const retryParams = merge(Object.create(null), this.defaultWaitParams, params)
```

if we want this